### DISCUSSION TITLE
Can't install Tinypilot

### DISCUSSION THREAD

**Discussion by yusisushi** [tiny-pilot/tinypilot#1762] [General]
Hi, it seems no mather what, I cannot get tinypilot installed.

I always end up with the same code
Command I run (from simple installation);
```
curl \
  --silent \
  --show-error \
  https://raw.githubusercontent.com/tiny-pilot/tinypilot/master/get-tinypilot.sh | \
    bash - && \
  sudo reboot
```

output
```
...
+ HTTP_CODE=000
+ clean_up
+ sudo umount --lazy /mnt/tinypilot-installer
+ sudo rm -rf /opt/tinypilot-updater /mnt/tinypilot-installer
```

I have the RBPv4 with 8 gigs of ram. Initially I had Debian 12 64bit installed but now I clean-installed Raspberry Pi OS Bullseye (32-bit) with the same results.

I'm confused, the readme literally says "[Raspberry Pi 4](https://smile.amazon.com/dp/B07TD42S27) (all variants work)"

**Comment by db39** [comment]
Hi @yusisushi, I'm sorry you're running into issues installing TinyPilot on your Raspberry Pi.

I've just tried to reproduce your issue with a fresh installation of Raspberry Pi OS Lite Bullseye (32-bit), but I didn't hit the same issue. The amount of RAM your device has shouldn't affect the installation.

Could you share the full output with me to review?

**Reply by yusisushi** [reply]
Ok, I'll do that. Just to confirm my OS;

```
 cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

I did apt update && upgrage before executing below as well, other than that its a clean install

It seems to just quit without any special warnings
```
yuska@rbp:~ $ curl \
  --silent \
  --show-error \
  https://raw.githubusercontent.com/tiny-pilot/tinypilot/master/get-tinypilot.sh | \
    bash - && \
  sudo reboot
+ HAS_PRO_INSTALLED=0
++ dirname bash
+ SCRIPT_DIR=.
+ [[ . = \. ]]
+ SCRIPT_DIR=/opt/tinypilot
+ readonly SCRIPT_DIR
+ readonly TINYPILOT_README=/opt/tinypilot/README.md
+ TINYPILOT_README=/opt/tinypilot/README.md
+ [[ -f /opt/tinypilot/README.md ]]
+ readonly HAS_PRO_INSTALLED
+ [[ 0 = 1 ]]
+ readonly LEGACY_INSTALLER_DIR=/opt/tinypilot-updater
+ LEGACY_INSTALLER_DIR=/opt/tinypilot-updater
+ readonly RAMDISK_SIZE_MIB=560
+ RAMDISK_SIZE_MIB=560
++ free --mebi
++ tr --squeeze-repeats ' '
++ grep --fixed-strings Mem:
++ cut --delimiter ' ' --fields 7
+ AVAILABLE_MEMORY_MIB=7621
+ readonly AVAILABLE_MEMORY_MIB
+ INSTALLER_DIR=/mnt/tinypilot-installer
+ trap clean_up EXIT
+ ((  7621 >= 560  ))
+ sudo mkdir /mnt/tinypilot-installer
+ sudo mount --types tmpfs --options size=560m --source tmpfs --target /mnt/tinypilot-installer --verbose
mount: tmpfs mounted on /mnt/tinypilot-installer.
+ readonly INSTALLER_DIR
+ readonly TMPDIR=/mnt/tinypilot-installer/tmp
+ TMPDIR=/mnt/tinypilot-installer/tmp
+ export TMPDIR
+ sudo mkdir /mnt/tinypilot-installer/tmp
+ readonly BUNDLE_FILE=/mnt/tinypilot-installer/bundle.tgz
+ BUNDLE_FILE=/mnt/tinypilot-installer/bundle.tgz
++ curl https://gk.tinypilotkvm.com/community/download/latest --location --output /mnt/tinypilot-installer/bundle.tgz --write-out '%{http_code}' --silent


+ HTTP_CODE=000
+ clean_up
+ sudo umount --lazy /mnt/tinypilot-installer
+ sudo rm -rf /opt/tinypilot-updater /mnt/tinypilot-installer
yuska@rbp:~ $
```
It seems like the below url is unavailable for me, which becomes evident when I leave out the `--silent` parameter in the curl command
https://gk.tinypilotkvm.com/community/download/latest
```
curl https://gk.tinypilotkvm.com/community/download/latest --location --output /mnt/tinypilot-installer/bundle.tgz --write-out '%{http_code}'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:01:21 --:--:--     0
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to gk.tinypilotkvm.com:443
```

I seem to get this working on different physical locations though. So it seems there is an internet routing issue for me. Unfortunately rebooting my firewall and obtaining a different WAN IP does not resolve the issue.

I will try to download the script needed locally and report back.

Update: got it working by using alternate internet connection... Thanks for taking time to try and reproduce the issue anyway!

**Reply by db39** [reply]
Thanks for the update, @yusisushi. I'm glad you're now up and running!
